example/reddit_example.py

```diff
@@ -0,0 +1,50 @@
+import logging
+import sys
+import time
+from datetime import datetime, timedelta
+
+import pytz
+
+from obsei.misc.utils import DATETIME_STRING_PATTERN
+from obsei.source.reddit import RedditConfig, RedditSource
+from obsei.workflow.store import WorkflowStore
+from obsei.workflow.workflow import Workflow, WorkflowConfig
+
+
+def print_state(id: str):
+    logger.info(f'Source State: {source.store.get_source_state(id)}')
+
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+since_time = datetime.utcnow().astimezone(pytz.utc) + timedelta(hours=-2)
+# Credentials will be fetched from env variable named reddit_client_id and reddit_client_secret
+source_config = RedditConfig(
+    subreddits=["wallstreetbets"],
+    lookup_period=since_time.strftime(DATETIME_STRING_PATTERN)
+)
+
+source = RedditSource(store=WorkflowStore())
+
+workflow = Workflow(
+    config=WorkflowConfig(
+        source_config=source_config,
+    ),
+)
+source.store.add_workflow(workflow)
+
+
+for i in range(1, 4):
```

Comment:
@lalitpagaria  Could you help me please understand the significance of this range? Is it for retries?

Reply:
@Saga4 Sorry for the late reply.  The range is not significant here.
It just shows how one can maintain a state while using Reddit observer.
Following code create DB table (by default SQLite) and store state data like tweet id, time, comment id, etc. So when you run the same script next time Obsei will not pull duplicate data.
```
workflow = Workflow(
    config=WorkflowConfig(
        source_config=source_config,
    ),
)
source.store.add_workflow(workflow)
```